csrc/multidevice/device_mesh.h

```diff
@@ -18,39 +21,34 @@ namespace nvfuser {
 // will add support for n-dimensional meshes.
 class DeviceMesh final {
  public:
-  DeviceMesh(std::vector<DeviceIdxType> devices = {}) {
-    setDevices(devices);
-  }
-
-  DeviceMesh& operator=(const std::vector<DeviceIdxType>& devices) {
-    setDevices(devices);
-    return *this;
-  }
+  explicit DeviceMesh(std::vector<DeviceIdxType> devices = {});
+  DeviceMesh(std::initializer_list<DeviceIdxType> devices);
+  DeviceMesh(const DeviceMesh&) = default;
+  DeviceMesh(DeviceMesh&&) = default;
+  DeviceMesh& operator=(const DeviceMesh&) = default;
+  DeviceMesh& operator=(DeviceMesh&&) = default;
 
   // Creates a device mesh of [0 .. num_devices-1]. I didn't make it a
   // constructor because single-element initializer lists would be directed to
   // use that instead of the constructor for vectors.
   static DeviceMesh createForNumDevices(int64_t num_devices);
 
-  std::string toString() const;
```

Comment:
I inlined its definition to python_bindings.cpp, the only use case of this function. C++ code should find `operator<<` more useful, so I'm removing this for now. However, I'm happy to add it back if/when it becomes useful in the future.